### PR TITLE
microservice init

### PR DESCRIPTION
**Pull request by yuhao900914**

### CODE REVIEW

File: demo/src/com/amplitude/Demo.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+public class Demo {
+
```
Comment (haoliu-amp) [inline demo/src/com/amplitude/Demo.java:13]: extra line

File: demo/src/com/amplitude/Demo.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	public static void main(String[] args) {
+		SpringApplication.run(Demo.class, args);
+	}
+
+	@Bean
+	public CommandLineRunner commandLineRunner(ApplicationContext ctx) {
+		return args -> {
+
+			System.out.println("Let's inspect the beans provided by Spring Boot:");
```
Comment (haoliu-amp) [inline demo/src/com/amplitude/Demo.java:22]: Could we change this to something meaningful?

File: demo/src/com/amplitude/Demo.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+			String[] beanNames = ctx.getBeanDefinitionNames();
+			Arrays.sort(beanNames);
+			for (String beanName : beanNames) {
+				System.out.println(beanName);
+			}
+
```
Comment (haoliu-amp) [inline demo/src/com/amplitude/Demo.java:29]: extra line.

File: demo/src/com/amplitude/Demo.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+		};
+	}
+
```
Comment (haoliu-amp) [inline demo/src/com/amplitude/Demo.java:32]: same

**Review by haoliu-amp — CHANGES_REQUESTED** [review]
src/com/amplitude/Amplitude.java → app/src/com/amplitude/Amplitude.java 

This is not right. 
Should be something like:
main/java/com/amplitude/xxx.java
demo/java/com/amplitude/xxx.java
test/java/com/amplitude/xxx.java

File: demo/build.gradle [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    module {
+        sourceDirs += file('src')
+    }
+}
+
+sourceSets.main.java.srcDirs = ['src']
+
+dependencies {
+    implementation 'org.json:json:20201115'
+    implementation 'org.springframework.boot:spring-boot-starter-web'
```
Comment (haoliu-amp) [inline demo/build.gradle:31]: What's this package about? A package for starter?

File: demo/build.gradle [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+application {
+    mainClassName = "com.amplitude.Demo"
+}
+
+task runWithJavaExec(type: JavaExec) {
+    group = "Execution"
+    description = "Run the main class with JavaExecTask"
+    classpath = sourceSets.main.runtimeClasspath
+    main = "com.amplitude"
```
Comment (haoliu-amp) [inline demo/build.gradle:42]: This is going to run the main method? Should this be `com.amplitude.Demo`? Also this is a micro service, why you bother running main?

**Review by haoliu-amp — CHANGES_REQUESTED** [review]

File: demo/build.gradle (hunk 5 above)
Comment (yuhao900914) [reply demo/build.gradle:31]: That's the required dependency for `Spring Web`.  A starter for building web.

**Comment by dantetam** [comment]
@yuhao900914 Did you test in IntelliJ how the two modules interact with each other? Are there any issues marking both `app` and `demo` both as source folders?

File: demo/build.gradle (hunk 6 above)
Comment (dantetam) [reply demo/build.gradle:42]: com.Amplitude.Demo was just for our internal testing of the API with `public static void main(String[] args)`

Ideally, yes, we would like a Gradle job like this one where the microservices app runs, sets up an endpoint that calls the Java SDK logEvent.

**Comment by dantetam** [comment]
@haoliu-amp @yuhao900914 

Yes, the folder structure should be

src <---- Marked as src folder by gradle
      main
      demo
      test

File: src/demo/java/com/amplitude/DemoController.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package com.amplitude;
+
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@RestController
+public class DemoController {
+	@RequestMapping("/")
+	public String index() {
+		return "Amplitude Java SDK Demo!";
```
Comment (haoliu-amp) [inline src/demo/java/com/amplitude/DemoController.java:10]: Should try to log one event here.

**Review by haoliu-amp — CHANGES_REQUESTED** [review]
Look much better!

**Review by haoliu-amp — APPROVED** [review]

**Review by dantetam — APPROVED** [review]

**Pull request merged**